### ISSUE TITLE
local data for widgets

### ISSUE THREAD

**Issue by AliciaSchep** [vegawidget/vegawidget#59]
Giving a local path to a spec doesn't work when rendering the widget
```
data_path <- system.file("example-data/seattle-weather.csv", package = "vegawidget")
spec_precip <-
    list(
      `$schema` = vega_schema(),
      data = list(url = data_path),
      mark = "tick",
      encoding = list(
        x = list(field = "precipitation", type = "quantitative")
      )
    ) %>%
    as_vegaspec()
vegawidget(spec_precip)
```
However, the following does work:

```r
data_path <- system.file("example-data/seattle-weather.csv", package = "vegawidget")
filedep <- htmltools::htmlDependency(     
  name = "data",    
  version = "0.1.0", 
  src = c(file = file.path(dirname(data_path))), 
  attachment = basename(data_path), 
  all_files = FALSE 
)
spec_precip <-
    list(
      `$schema` = vega_schema(),
      data = list(url = 'lib/data-0.1.0/seattle-weather.csv'),
      mark = "tick",
      encoding = list(
        x = list(field = "precipitation", type = "quantitative")
      )
    ) %>%
    as_vegaspec()
vegawidget(spec_precip, dependencies = filedep)
```
(Thanks to https://community.rstudio.com/t/including-dynamically-created-files-into-htmlwidgets-dependencies/3453/4  for the solution) 

This makes me think that we could have something like a 'local_file' argument to vegawidget in order to do all of that work internally (including setting a baseUrl in the JS so that the spec doesn't have to 'lib/data-0.1.0')

**Comment by ijlyttle** [comment]
Nice catch!

**Comment by AliciaSchep** [comment]
Closed in #61, will make new issues for improvements!
